Client/Frontend/Sync/SyncSettingsTableViewController.swift

```diff
@@ -89,23 +98,82 @@ class SyncSettingsTableViewController: UITableViewController {
     // MARK: - Table view methods.
     
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        // Only button section is tappable.
-        if indexPath.section != Sections.buttons.rawValue { return }
-        
-        tableView.deselectRow(at: indexPath, animated: true)
+        defer { tableView.deselectRow(at: indexPath, animated: true) }
         
-        switch indexPath.row {
-        case ButtonsSection.Position.addAnotherDevice.rawValue:
+        if indexPath.section == Sections.buttons.rawValue {
             addAnotherDeviceAction()
-        case ButtonsSection.Position.removeDevice.rawValue:
-            removeDeviceAction()
-        default:
-            log.error("IndexPath row for SyncDeviceList buttons out of range")
+            return
         }
+        
+        guard let frc = frc, let deviceCount = frc.fetchedObjects?.count else { return }
+        let device = frc.object(at: indexPath)
+        
+        let actionShet = UIAlertController(title: device.name, message: nil, preferredStyle: .actionSheet)
+        
+        let removeAction = UIAlertAction(title: Strings.SyncRemoveDeviceAction, style: .destructive) { _ in
+            var alertType: DeviceRemovalType?
+            
+            if deviceCount == 1 {
+                alertType = .lastDeviceLeft
+            } else if device.isCurrentDevice {
+                alertType = .currentDevice
+            } else {
+                alertType = .otherDevice
```

Comment:
I think a bit cleaner here is not using `else` but instead setting alertType explicitly:

`var alertType = DeviceRemovalType.otherDevice // notice, no optional`

Then overwrite via the if-else.

This way it is also not optional so you don't need to unwrap it then.